lib/zendesk_apps_support/validations/manifest.rb

```diff
@@ -59,7 +57,56 @@ def collate_manifest_errors(package)
 
           errors.flatten.compact
         end
-        # rubocop:enable Metrics/AbcSize
+
+        def type_checks(manifest)
+          errors = []
+          errors << boolean_error(manifest)
+          errors << string_error(manifest)
+          errors << no_template_format_error(manifest)
+          unless manifest.experiments.is_a?(Hash)
+            errors << ValidationError.new(
+              :unacceptable_hash,
+              field: 'experiments',
+              value: manifest.experiments.class.to_s
+            )
+          end
+          whitelist = manifest.domain_whitelist
+          unless whitelist.nil? || whitelist.is_a?(Array) && whitelist.all? { |dom| dom.is_a? String }
+            errors << ValidationError.new(:unacceptable_array_of_strings, field: 'domainWhitelist')
+          end
+          parameters = manifest.original_parameters
+          unless parameters.nil? || parameters.is_a?(Array)
+            errors << ValidationError.new(
+              :unacceptable_array,
+              field: 'parameters',
+              value: parameters.class.to_s
+            )
+          end
+          errors
+        end
+
+        def string_error(manifest)
+          errors = []
+          manifest_strings = %i(
+            default_locale
+            version
+            framework_version
+            remote_installation_url
+            terms_conditions_url
+            google_analytics_code
+          )
+          errors.concat(manifest_strings.map do |field|
```

Comment:
finding this a bit ugly/unreadable. could do:
```
errors << manifest_strings.map do |field|
  validate_string(manifest.public_send(field), field)
end
```

and don't worry about the .compact, (since collate_manifest_errors does that anyway) or do it at the end of the function